pkgs/native_assets_cli/lib/src/api/link_output.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'architecture.dart';
+import 'asset.dart';
+import 'build_output.dart';
+import 'os.dart';
+
+extension type LinkOutput(BuildOutputImpl _buildOutput) {
```

Comment:
`BuildOutputImpl` should not occur in the public API for users.

Can this be an extension type on `BuildOutput` instead?